aptos-move/framework/aptos-framework/sources/account/common_account_abstractions/derivable_account_abstraction_solana.move

```diff
@@ -0,0 +1,251 @@
+/// Domain account abstraction using ed25519 hex for signing.
+///
+/// Authentication takes digest, converts to hex (prefixed with 0x, with lowercase letters),
+/// and then expects that to be signed.
+/// authenticator is expected to be signature: vector<u8>
+/// account_identity is raw public_key.
+module aptos_framework::derivable_account_abstraction_ed25519_hex {
+    use aptos_framework::auth_data::AbstractionAuthData;
+    use aptos_std::ed25519::{
+        Self,
+        new_signature_from_bytes,
+        new_unvalidated_public_key_from_bytes,
+    };
+    use std::bcs;
+    use std::chain_id;
+    use std::error;
+    use std::transaction_context;
+
+    const EINVALID_SIGNATURE: u64 = 1;
+    const EINVALID_BASE_58_PUBLIC_KEY: u64 = 2;
+    const EUNSUPPORTED_CHAIN_ID: u64 = 3;
+    const EMISSING_ENTRY_FUNCTION_PAYLOAD: u64 = 4;
+
+    const BASE_58_ALPHABET: vector<u8> = b"123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz";
+    const HEX_ALPHABET: vector<u8> = b"0123456789abcdef";
+
+    fun split_abstract_public_key(abstract_public_key: &vector<u8>): (vector<u8>, vector<u8>) {
+        // First 44 bytes are the base58 utf8 encoded public key
+        let base58_public_key = abstract_public_key.slice(0, 44);
+        let domain = abstract_public_key.slice(44, abstract_public_key.length());
+        (base58_public_key, domain)
+    }
+
+    fun network_name(): vector<u8> {
+        let chain_id = chain_id::get();
+        if (chain_id == 1) {
+            b"mainnet"
+        } else if (chain_id == 2) {
+            b"testnet"
+        } else {
+            abort(EUNSUPPORTED_CHAIN_ID)
```

Comment:
we also support devnet?
also, can we add `if (chain_id == 4)` for `localnet`?

Reply:
How about something like this?
https://github.com/aptos-labs/aptos-wallet-adapter/blob/ad3b6bbde68fb8b994f7c6ff3ab03081e903af52/packages/derived-wallet-base/src/envelope.ts#L12-L21

Basically, for known networks we should the name, for everything else just the chain id?

Reply:
> Basically, for known networks we show the name, for everything else just the chain id?

I'm not completely sold on just returning the chain ID. Part of the purpose of showing the network name is to let the user know that their transaction is happening on prod/dev

Reply:
I highly support in showing the actual network name, it is super helpful for dev exp as well as for user exp. Also, I dont think we should "encourage" it to be used on non-aptos/different networks and if so they should implement a custom authentication function implementation.

In the adapter, we query for the devnet chain id so it is up to date with the current id 
https://github.com/aptos-labs/aptos-wallet-adapter/blob/ad3b6bbde68fb8b994f7c6ff3ab03081e903af52/packages/derived-wallet-solana/src/SolanaDerivedWallet.ts#L206

We can however, to make it more align with the adapter code and "flexible", we dont `abort` for unsupported chain id, but just whatever the number is

Reply:
@0xmaayan - for mainnet/ testnet we should use network name. 

For others, I'm not sure move code knows what is the network name - i e. Whether it is localnet or devnet (which changes with every release) or whether it is is another temporarily run network - like previewnet or randomnet, etc.

We also don't want txn from one denver run (with denver 167) to be replayable with next - denver 168.

So I think best we can do is `mainnet` / `testnet` / `developer network #167` .

We can ask on #blochain-dev to confirm

Reply:
synced with Brian offline, we actually build the message here and dont have it so we can't regex or know the network name.

We agreed on using 1 for mainnet, 2 for testnet and 4 for local (that would help a lot with debugging and developing), for all other we use devnet - and if it is not devnet the verification fails and thats ok as we only expect to support aptos networks

Reply:
first we need to distinguish chain_id for different devnets, so it needs to be `devnet (#164)`. Second it would be very confusing if we run a specialized network - like previewnet or randomnet or some other product-net, and have it show as devnet. 

so I suggest using `developer network (#164)` instead of devnet, so it's not incorrect when it is not a devnet

Reply:
if it's not mainnet/testnet, what do we think about just excluding the network name from the message?

Reply:
I think we should include something, so using `developer network` works for me. Btw, what does `#164` represent?

Reply:
we need to include for replay protection. #164 is just chain id

so if chain_id = 164, we write `developer network (#164)`
if chain_id = 130, we write `developer network (#130)`

Every week, devnet gets a new number, and we recycle after we use all, and we get different numbers for previewnet, randomnet, etc

Reply:
we will always have replay protection for mainnet/testnet even if we exclude the chain ID from non-(mainnet/testnet). I don't think it's as important to have replay protection for other networks.

Reply:
Catching up only now sorry.

I agree that replay-ability shouldn't be an issue since the chain id is included in the signing message and will affect its digest, regardless of whether it's displayed.

I think that for "stable" networks like Mainnet, Testnet and Localnet we're all on the same page and we should display their name.

Whether we display the chain id, "development network" or nothing at all in all other cases is up to y'all I don't feel strongly about it. I would prefer showing something rather than nothing though.